File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int n = prices.size(), minSoFar = prices[0], res = 0;

        for(int i=1; i<n; i++){
            res = max(res, prices[i]-minSoFar);
            minSoFar = min(minSoFar, prices[i]);
        }

        return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:  
        n = len(s)
        l, r = 0, 0
        m = {}
        res = 0
        
        while r < n:
            
            if s[r] in m:
                l = max(m[s[r]] + 1, l)
                
            res = max(res, r-l+1)

            m[s[r]] = r
            r += 1

        return res

            
